consensus/src/threshold_simplex/actors/batcher/actor.rs

```diff
@@ -459,10 +442,17 @@ impl<
                         }) => {
                             current = new_current;
                             finalized = new_finalized;
-                            let leader_index = self.supervisor.is_participant(current, &leader).unwrap();
-                            work.entry(current)
-                                .or_insert(Round::new(self.blocker.clone(), self.reporter.clone(), self.supervisor.clone(), self.inbound_messages.clone(), current, initialized))
-                                .set_leader(leader_index);
+                            let leader_index = self.participants.signer_index(&leader).unwrap();
+                            work.entry(current).or_insert(
+                                Round::new(
+                                    self.participants.clone(),
+                                    self.signing.clone(),
+                                    self.blocker.clone(),
+                                    self.reporter.clone(),
+                                    self.inbound_messages.clone(),
+                                    initialized
+                                )
```

Comment:
nit: since this `Round::new(...)` is duplicated a few times, consider that a helper method on `self` could reduce some duplication